File: kolko2/python/cukiernia/main.py
def cukiernia(il_przed, maks_suma_wag, waga, wartosc):
    tablica_wynikow = [
        [0 for i in range(maks_suma_wag + 1)] for j in range(il_przed + 1)
    ]
    for i in range(1, il_przed + 1):
        p = i - 1
        for j in range(1, maks_suma_wag + 1):
            if waga[p] <= j:
                tablica_wynikow[i][j] = max(
                    wartosc[p] + tablica_wynikow[p][j -
                                                    waga[p]], tablica_wynikow[p][j]
                )
            else:
                tablica_wynikow[i][j] = tablica_wynikow[p][j]
    return tablica_wynikow[il_przed][maks_suma_wag]


def main():
    il_przed, maks_suma_wag = list(map(int, input().split()))
    waga = []
    wartosc = []
    for i in range(il_przed):
        wejsc = input().split()
        waga.append(int(wejsc[0]))
        wartosc.append(int(wejsc[1]))
    print(cukiernia(il_przed, maks_suma_wag, waga, wartosc))


if __name__ == "__main__":
    main()

File: kolko2/cpp/cukiernia/main.cpp
#include <iostream>
using namespace std;

int max(int x, int y) { return (x > y) ? x : y; }

int ciastkowybieracz(int pieniadze, int il_przedmiotow, int masa[],
                     int wartosc[]) {
    int i, wt;
    int tablica_wynikow[il_przedmiotow + 1][pieniadze + 1];
    for (i = 0; i <= il_przedmiotow; i++) {
        for (wt = 0; wt <= pieniadze; wt++) {
            if (i == 0 || wt == 0)
                tablica_wynikow[i][wt] = 0;
            else if (wartosc[i - 1] <= wt)
                tablica_wynikow[i][wt] = max(
                    masa[i - 1] + tablica_wynikow[i - 1][wt - wartosc[i - 1]],
                    tablica_wynikow[i - 1][wt]);
            else
                tablica_wynikow[i][wt] = tablica_wynikow[i - 1][wt];
        }
    }
    return tablica_wynikow[il_przedmiotow][pieniadze];
}

int main() {
    int il_przedmiotow, pieniadze;
    cin >> il_przedmiotow >> pieniadze;
    int masa[il_przedmiotow], wartosc[il_przedmiotow];
    for (int i = 0; i < il_przedmiotow; i++) {
        cin >> wartosc[i] >> masa[i];
    }
    cout << ciastkowybieracz(pieniadze, il_przedmiotow, masa, wartosc) << "\n";
}